File: sdk-java/src/test/java/io/superdurable/dex/iwfcompat/NoStartStateTest.java
/*
 * Copyright (c) 2026 Super Durable, Inc.
 *
 * Licensed under the Super Durable Source License 1.0.
 * You may not use this file except in compliance with the License.
 * See the LICENSE file in the repository root.
 *
 * SPDX-License-Identifier: LicenseRef-Super-Durable-1.0
 */

package io.superdurable.dex.iwfcompat;

import io.superdurable.dex.Client;

public final class NoStartStateTest {
    void compileNoStartStep(final Client client) {
        client.startFlow(IwfFlows.NO_START, "no-start", null);
        final NoStartFlow stub = client.newRpcStub(
                NoStartFlow.class,
                "no-start");
        final Long output = client.invokeRPC(stub::invoke, "input");
        consume(output);
    }

    void compileNoStep(final Client client) {
        client.startFlow(IwfFlows.NO_STATE, "no-step", null);
        final NoStateFlow stub = client.newRpcStub(
                NoStateFlow.class,
                "no-step");
        final Integer output = client.invokeRPC(stub::increaseCounter);
        client.stopFlow("no-step");
        consume(output);
    }

    void compileDeadEnd(final Client client) {
        client.startFlow(IwfFlows.DEAD_END, "dead-end", null);
        final DeadEndFlow stub = client.newRpcStub(
                DeadEndFlow.class,
                "dead-end");
        final Integer size = client.invokeRPC(stub::publishInternal);
        consume(size);
    }

    private static void consume(final Object value) {
    }
}

File: sdk-python/tests/integ/no_start_state_test.py
# Portions of this file are derived from indeedeng/iwf-java-sdk.
# Those portions are licensed under the Apache License, Version 2.0.
# See LICENSES/Apache-2.0.txt and LEGACY_NOTICES.md.
#
# Modifications Copyright (c) 2026 Super Durable, Inc.
#
# Modifications are licensed under the Super Durable Source License 1.0.
# Third-Party Materials remain under the Apache License, Version 2.0.
# See LICENSE and LEGACY_NOTICES.md.

from dex import Client

from .dead_end_flow import DeadEndFlow
from .no_start_flow import NoStartFlow
from .no_state_flow import NoStateFlow


def compile_no_start_step(client: Client) -> None:
    flow = NoStartFlow()
    client.start_flow(flow, "no-start", None)
    output: int = client.invoke_rpc(flow.invoke, "no-start", "input")
    del output


def compile_no_step(client: Client) -> None:
    flow = NoStateFlow()
    client.start_flow(flow, "no-step", None)
    output: int = client.invoke_rpc(flow.increase_counter, "no-step")
    client.stop_flow("no-step")
    del output


def compile_dead_end(client: Client) -> None:
    flow = DeadEndFlow()
    client.start_flow(flow, "dead-end", None)
    size: int = client.invoke_rpc(flow.publish_internal, "dead-end")
    del size
